### DISCUSSION TITLE
Default Table Configuration Layout

### DISCUSSION THREAD

**Discussion by bribeck** [netbox-community/netbox#9184] [General]
I have created a table config layout as super user is there a way to have all current and future users use this table layout? This way everybody is the seeing the same table details.

**Comment by candlerb** [comment]
What Netbox version are you using? In the [release notes](https://github.com/netbox-community/netbox/releases/tag/v3.2.0) for v3.2.0:

-----
### Improved User Preferences (https://github.com/netbox-community/netbox/issues/7759)
A robust new mechanism for managing user preferences is included in this release. The user preferences form has been improved for better usability, and administrators can now define default preferences for all users with the [`DEFAULT_USER_PREFERENCES`](https://github.com/netbox-community/netbox/blob/develop/docs/configuration/dynamic-settings.md#default_user_preferences) configuration parameter. For example, this can be used to define the columns which appear by default in a table:

```
DEFAULT_USER_PREFERENCES = {
    'tables': {
        'IPAddressTable': {
            'columns': ['address', 'status', 'created', 'description']
        }
    }
}
```

Users can adjust their own preferences under their user profile. A complete list of supported preferences is available in NetBox's [developer documentation](https://github.com/netbox-community/netbox/blob/develop/docs/development/user-preferences.md).

-----
(Aside: the links in the original release notes are broken, but I have corrected them above)

**Reply by ybizeul** [reply]
Works great for new users, but it's a shame the defined preferences aren't used when user *Reset* the table

**Reply by candlerb** [reply]
Are they not? Which Netbox version are you running?

I would expect "Reset" to revert the table to the configured defaults.  What does it do instead?

**Reply by ybizeul** [reply]
This is what I have in `netbox.yaml` :
```
DEFAULT_USER_PREFERENCES: {"tables":{"DeviceTable":{"columns":["name","rack","device_role","manufacturer","device_type","primary_ip","comments"]}}}
```

When the user has just been created and displays that table, I get those exact columns.

Now if I reset it, I get the following columns again :
```
Name | Status | Tenant | Site | Location | Rack | Role | Manufacturer | Type | IP Address
```
This is netbox-bbd86d8f5-7tnwj (v3.2.8)

**Reply by candlerb** [reply]
I have tested with v3.5.7, and I see the same as you: after resetting the table, I get the columns you see rather than those from DEFAULT_USER_PREFERENCES. If I then go to `/user/preferences/` I see an empty list of columns for DeviceTable: and I would expect this to mean DEFAULT_USER_PREFERENCES rather than some original hard-coded defaults. I think you could raise this as a bug.

Looking at the source for v3.5.7:

* `netbox/utilities/templates/helpers/table_config_form.html` creates a button with ID `reset_tableconfig`
* there's no handler bound directly to that ID AFAICS, but the submit handler in `netbox/project-static/src/tableConfig.ts` tests for this button:

    ```
      const reset = document.activeElement?.getAttribute('value') === 'Reset';
    ```

* if `reset` is true, the code sets the list of columns to empty

**Reply by ybizeul** [reply]
Opened issue https://github.com/netbox-community/netbox/issues/13312

**Reply by ybizeul** [reply]
@candlerb Are you using https://github.com/bootc/netbox-chart or a standard deployment ?

**Reply by candlerb** [reply]
Standard, non-docker deployment.
